### DISCUSSION TITLE
Is the call wrong?

### DISCUSSION THREAD

**Discussion by Galllaa** [ramokz/phantom-camera#101] [General]
Hi,  here is a short clip of the problem i have. 

https://github.com/ramokz/phantom-camera/assets/105984870/ae9a366b-c706-4c06-b41d-c8568625d39a

**Comment by Galllaa** [comment]
Well i found whats causing this. If you instead of this `@onready var cameraLeft : PhantomCamera2D = $CameraLeft`  make this  `@onready var cameraLeft = $CameraLeft` there are no errors.

**Comment by ramokz** [comment]
Had a look at this and it's an interesting one.
Like you said, removing the type in the variable declaration removes the error. Obviously, that is A) not ideal and B) not solving the problem. 

Doing some more tests, I realized that some properties are accessible using code, and some aren't. For example, `zoom`, `priority` and `inactive_update_mode` are accessible using `cameraLeft.priority`. Not sure why the `Properties` bit isn't needed, nor why only some properties are available.

One solution is to have additional helper functions to access the properties. There are a few already, such as `cameraLeft.get_priority()`, so what I can do is add setters/getter functions to the remainder ones - including `is_active`.

Until I submit a release with that, you can add the following to the bottom of the `phantom_camera_2D.gd` plugin script (inside `addons/phantom_camera/scripts/phantom_camera`):
``` gdscript
func is_active() -> bool:
	return Properties.is_active
```
Then you can get the property using `cameraLeft.is_active()`.
I will be using the same method name in that release, so upgrading the plugin later on won't break anything.

**Reply by ramokz** [reply]
Have added an issue for this #102, and made a start on it as well.

**Reply by Galllaa** [reply]
Thats great. Thank you :)

**Comment by ramokz** [comment]
Closing the conversation as this feature is now fully supported in v0.4.
